src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -4596,10 +4596,9 @@ private StringBuilder localizedMagnitude(Formatter fmt, StringBuilder sb,
             }
 
             // apply zero padding
-            if (width != -1 && Flags.contains(f, Flags.ZERO_PAD)) {
-                for (int k = sb.length(); k < width; k++) {
-                    sb.insert(begin, zero);
-                }
+            if (width > sb.length() && Flags.contains(f, Flags.ZERO_PAD)) {
+                String zeros = Character.toString(zero).repeat(width - sb.length());
```

Comment:
`Character.toString(zero)` could be replaced with `String.valueOf(zero)` saving one more method call